Filter JSON RPC crash due to nil timeout filter

## Filter JSON RPC crash due to not concurrency safe map

### Description

The timeout filter might be nil, but it is not checked carefully.

### Your environment

* Mac OS 12
* Jury v0.4.2

### Steps to reproduce

* Create a local dev mode server, start it.
* Use MetaMask to connect this network.
* Use some WebSocket client tool like `Smart Websocket Client` to connect the server.
* Subscribe some block changes (`eth_newBlockFilter`)
* Wait a few seconds (30 seconds or so), and uninstall some of them (`eth_uninstallFilter`).
* Use MetaMask to send some transactions.

Though it might not happen 100%, but it is easy to reproduce.

### Expected behaviour

Should not crash whatever a WebSocket client do.

### Actual behaviour

Crash.

### Logs

```
2022-05-23T15:10:32.019+0800 [INFO]  jury.consensus.ibft.acceptState: we are the proposer: block=2388
2022-05-23T15:10:32.019+0800 [INFO]  jury.consensus.ibft: executed txns: failed =0 successful=0 remaining in pool=0
2022-05-23T15:10:32.020+0800 [INFO]  jury.consensus.ibft: build block: number=2388 txns=0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x54fd597]

goroutine 135 [running]:
github.com/dogechain-lab/jury/jsonrpc.(*FilterManager).Run(0xc000e2ab40)
	/Users/Darian/go/src/github.com/dogechain-lab/jury/jsonrpc/filter_manager.go:296 +0x277
created by github.com/dogechain-lab/jury/jsonrpc.newDispatcher
	/Users/Darian/go/src/github.com/dogechain-lab/jury/jsonrpc/dispatcher.go:57 +0x206
```

### Proposed solution

Fix it.

--- a/jsonrpc/filter_manager.go
+++ b/jsonrpc/filter_manager.go
@@ -275,6 +275,7 @@ func (f *FilterManager) Run() {
 
 	var timeoutCh <-chan time.Time
 
+OUT_LOOP:
 	for {
 		// check for the next filter to be removed
 		filterBase := f.nextTimeoutFilter()
@@ -293,6 +294,13 @@ func (f *FilterManager) Run() {
 
 		case <-timeoutCh:
 			// timeout for filter
+			// might be nil
+			if filterBase == nil {
+				f.logger.Warn("timeout filterBase is nil")
+
+				continue OUT_LOOP
+			}
+
 			if !f.Uninstall(filterBase.id) {
 				f.logger.Error("failed to uninstall filter", "id", filterBase.id)
 			}

